.github/workflows/ci_rust.yml

```diff
@@ -75,16 +75,50 @@ jobs:
         # invoked on the `cargo test --all-features` pattern.
         run: RUST_LOG=TRACE cargo test --no-default-features --features pq
 
+  external-build-test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install Rust toolchain
+        id: toolchain
+        run: |
+          rustup toolchain install stable
+          rustup override set stable
+
+      - uses: camshaft/rust-cache@v1
+
+      - name: Checkout aws-lc
+        uses: actions/checkout@v4
+        with:
+          repository: aws/aws-lc
+          path: aws-lc
+
+      - name: Build aws-lc
+        working-directory: aws-lc
+        run: |
+          cmake -Bbuild \
+            -DBUILD_SHARED_LIBS=off \
+            -DBUILD_LIBSSL=off \
+            -DCMAKE_INSTALL_PREFIX=./install
+          cmake --build build -j $(nproc)
+          cmake --install build
+
       - name: Test external build
         # if this test is failing, make sure that api headers are appropriately
         # included. For a symbol to be visible in a shared lib, the
         # __attribute__((visibility("default"))) label must be on a declaration
         # in the same unit of compilation as the definition. Generally this just
         # means that if the linker can't resolve foo_method in tls/foo.c, you
         # forgot to include api/unstable/foo.h in tls/foo.c
-        if: ${{ matrix.os == 'ubuntu-latest' }}
         run: |
-          cmake . -Bbuild -DBUILD_SHARED_LIBS=on -DBUILD_TESTING=off
+          set -e
+
```

Comment:
Is set -e necessary here? I thought any run command failing always caused a failure. Otherwise, we'd need set -e everywhere.